roles/falcon_installation/tasks/api.yml

```
@@ -12,6 +12,14 @@
       content-type: application/x-www-form-urlencoded
   register: falcon_api_oauth2_token
 
+- name: CrowdStrike Falcon | Auto-discover CrowdStrike Cloud Region
+  ansible.builtin.set_fact:
+    falcon_cloud: "{{ item.value }}"
+  loop: "{{ lookup('dict', falcon_cloud_urls) }}"
+  when:
+    - falcon_cloud_autodiscover
+    - "falcon_api_oauth2_token.x_cs_region in item.key"
+
```

This is much faster and efficient. We should already have the `x_cs_region` so a simply lookup works. I through an extra when clause in case we don't get a region, but I'm not sure if that is a valid use case...
```suggestion
- name: CrowdStrike Falcon | Auto-discover CrowdStrike Cloud Region
  ansible.builtin.set_fact:
    falcon_cloud: "{{ falcon_cloud_urls[falcon_api_oauth2_token.x_cs_region] }}"
  when:
    - falcon_cloud_autodiscover
    - falcon_api_oauth2_token.x_cs_region != ""

```

---

@carlosmmatos done